src/main/java/org/terasology/metalrenegades/world/dynamic/MRBiome.java

```diff
@@ -100,10 +114,12 @@ public Block getBelowSurfaceBlock(Vector3ic pos, float density) {
     }
 
     private Block getStratum(Vector3ic pos) {
-        // Generate a strata index with a modulus of the y coordinate, adjusted with noise
-        // Strata are 20 (1/0.05) blocks high on average, but it varies since the noise value depends on height
-        float noise = strataNoise.noise(pos.x(), pos.y(), pos.z());
-        int idx = (int) Math.abs((float) pos.y() * 0.05 + noise * rocks.length + 1000) % rocks.length;
+        // Noise, adjusted to the interval [0,1]
+        float noise = strataNoise.noise(pos.x(), pos.y(), pos.z()) * 0.5f + 0.5f;
+        // Now pick an entry in `rocks` based on the noise
+        int idx = (int) (noise * rocks.length);
+        idx = Math.max(idx, 0);
+        idx = Math.min(idx, rocks.length - 1);
```

Comment:
I think you can use `org.joml.Math.clamp(min, max, val)` here:

```suggestion
        int idx = Math.clamp(0, rocks.length -1 , (int) (noise * rocks.length));
```